yax/refined/src/main/scala/doobie/refined/package.scala

```diff
@@ -0,0 +1,42 @@
+package doobie
+
+package object refined {
+
+  import eu.timepit.refined.api.{Refined, Validate}
+  import eu.timepit.refined.auto._
+  import eu.timepit.refined._
+  import scala.reflect.runtime.universe.TypeTag
+
+  import doobie.imports._
+
+  implicit def refinedMeta[T, P](implicit metaT: Meta[T], v: Validate[T, P], tag: TypeTag[T Refined P]): Meta[T Refined P] =
+    metaT.xmap[T Refined P](
```

Comment:
I think you need `nxmap` here because the values here can be null. There should be a test for read/write of `Option[A]` for some refined type `A` to make sure nullity is respected.